docs/design_of_gie.md

```diff
@@ -1,2 +1,143 @@
+  adapt to common query languages as much as possible.
+- The fragmentation of business scenarios: Different business scenarios have very different
+  requirements for graph queries. Some common scenarios are business intelligence scenarios,
+  high-throughput scenarios and their hybrid. In business intelligence scenarios, queries usually
+  involve more hops of graph traversal and complex relational computations such as aggregation,
+  deduplication and sorting. There is a higher demand for low latency of queries. In high-throughput
+  scenarios, queries are relatively simple and only perform one or two hops of graph traversal.
+  They value more the high throughput of queries, that is, to execute as many queries as possible
+  in a unit time. In addition, hybrid scenarios hope to get a better trade-off between low latency
+  and high throughput. The requirements for queries raised by different scenarios are fundamentally
+  the requirements for the underlying computing engines. To meet these requirements, it is inevitable
+  to introduce multiple computing engines
+
+:::{figure-md}
+
+<img src="images/states_of_gie.png"
+     alt="The current states of graph queries"
+     style="border-radius: 0.3125em; box-shadow: 0 2px 4px 0 rgba(34,36,38,.12),0 2px 10px 0 rgba(34,36,38,.08);"
+     width="65%">
+
+Figure 1. The current states of graph queries.
+:::
+
+As shown in Figure 1, in order to support as many query languages as possible and adapt to various
+application scenarios, a realistic graph query system inevitably has to face an engineering
+implementation problem with a total of “M×N”, that is, for each of the “M” query languages to
+interface with each of the “N” computing engines. And for each of these implementations, a whole
+process including language parsing, query validation, query optimization and runtime implementation
+needs to be completed. The whole process is lengthy and complicated. It is not easy to do well in
+one of these implementations, let alone do well in all these “M×N” combinations.
+
+## The Architecture
+
+Figure 2 shows the architecture of the GIE framework. It can be seen that the "M × N" fragmented
+engineering implementation in the original scene has been replaced by a unified GIE IR
+(Intermediate Representation) Layer. The GIE IR Layer is responsible for processing the two most
+core tasks in the graph query process, query verification and query optimization. Parsing for each
+language is completed in the corresponding client, and code generation for each execution engine is
+handled at the corresponding runtime. In this way, we only need to implement M parsers for M
+languages and N sets of code generation for N engines.
+
+:::{figure-md}
+
+<img src="images/design_of_gie.png"
+     alt="The architecture of GIE"
+     style="border-radius: 0.3125em; box-shadow: 0 2px 4px 0 rgba(34,36,38,.12),0 2px 10px 0 rgba(34,36,38,.08);"
```

Comment:
Just a suggestion: to make it consistent and concise, maybe remove the style to make figures flat and borderless.

Reply:
Okay. I refine the other images as well.